sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -1063,43 +1089,46 @@ pi_result piMemRelease(pi_mem Mem) {
   }
 
   if (--(Mem->RefCount) == 0) {
+    int Status = cm_support::CM_FAILURE;
+    std::lock_guard<std::mutex> Lock(Mem->Context->CmSurfaceManageLock);
```

Comment:
again, why do we need a lock here? And it is missing in the place where it is really needed - above line 1120